### ISSUE TITLE
Seg Fault on Exit in Example

### ISSUE THREAD

**Issue by kheaactua** [MASKOR/Qt3DPointcloudRenderer#2]
Building off master ( 0e06324da985441185523356518488fa6f7555b3 ) with `clang++-3.8`, Qt 5.7.0 on Ubuntu 14.04.03 I get a seg fault on exit with the example.

```
./Qt3DPointcloudRenderer_example 
Read Pointcloud "data/bunny.pcd" with 397 points.
Read Pointcloud "data/bunny_normal.pcd" with 397 points.
[1]    22423 segmentation fault (core dumped)  ./Qt3DPointcloudRenderer_example
```

The outputted core file tells me nothing, but I can get a little bit of info with clang's address sanitizer:

```
./Qt3DPointcloudRenderer_example 
Read Pointcloud "data/bunny.pcd" with 397 points.
Read Pointcloud "data/bunny_normal.pcd" with 397 points.
=================================================================
==23217==ERROR: AddressSanitizer: alloc-dealloc-mismatch (malloc vs operator delete) on 0x629000064200
    #0 0x51be90 in operator delete(void*) (/home/matt/workspace/qtquick/pointcloud/Qt3DPointcloudRenderer/bld/example/Qt3DPointcloudRenderer_example+0x51be90)
    #1 0x7f75d32c33c8  (/opt/Qt5.7.0/5.7/gcc_64/lib/libQt53DRender.so.5+0x15e3c8)
    #2 0x7f75d32c33f2  (/opt/Qt5.7.0/5.7/gcc_64/lib/libQt53DRender.so.5+0x15e3f2)
    #3 0x7f75d31f67d5  (/opt/Qt5.7.0/5.7/gcc_64/lib/libQt53DRender.so.5+0x917d5)
    #4 0x7f75d31f40cf in Qt3DRender::Render::NodeManagers::~NodeManagers() (/opt/Qt5.7.0/5.7/gcc_64/lib/libQt53DRender.so.5+0x8f0cf)
    #5 0x7f75d325649a in Qt3DRender::QRenderAspectPrivate::~QRenderAspectPrivate() (/opt/Qt5.7.0/5.7/gcc_64/lib/libQt53DRender.so.5+0xf149a)
    #6 0x7f75d32565f8 in Qt3DRender::QRenderAspectPrivate::~QRenderAspectPrivate() (/opt/Qt5.7.0/5.7/gcc_64/lib/libQt53DRender.so.5+0xf15f8)
    #7 0x7f75c83aa1f5 in QObject::~QObject() (/opt/Qt5.7.0/5.7/gcc_64/lib/libQt5Core.so.5+0x2a11f5)
    #8 0x7f75d3256258 in Qt3DRender::QRenderAspect::~QRenderAspect() (/opt/Qt5.7.0/5.7/gcc_64/lib/libQt53DRender.so.5+0xf1258)
    #9 0x7f75c83a364f in QObject::event(QEvent*) (/opt/Qt5.7.0/5.7/gcc_64/lib/libQt5Core.so.5+0x29a64f)
    #10 0x7f75c837a989 in QCoreApplication::notify(QObject*, QEvent*) (/opt/Qt5.7.0/5.7/gcc_64/lib/libQt5Core.so.5+0x271989)
    #11 0x7f75c837aadf in QCoreApplication::notifyInternal2(QObject*, QEvent*) (/opt/Qt5.7.0/5.7/gcc_64/lib/libQt5Core.so.5+0x271adf)
    #12 0x7f75c837d51a in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) (/opt/Qt5.7.0/5.7/gcc_64/lib/libQt5Core.so.5+0x27451a)
    #13 0x7f75c81b3806  (/opt/Qt5.7.0/5.7/gcc_64/lib/libQt5Core.so.5+0xaa806)
    #14 0x7f75c81b4600  (/opt/Qt5.7.0/5.7/gcc_64/lib/libQt5Core.so.5+0xab600)
    #15 0x7f75c78db183 in start_thread /build/eglibc-oGUzwX/eglibc-2.19/nptl/pthread_create.c:312
    #16 0x7f75c6fe537c in clone /build/eglibc-oGUzwX/eglibc-2.19/misc/../sysdeps/unix/sysv/linux/x86_64/clone.S:111

0x629000064200 is located 0 bytes inside of 16384-byte region [0x629000064200,0x629000068200)
allocated by thread T3 (Qt3DCore::QAspe) here:
    #0 0x4ed818 in __interceptor_malloc (/home/matt/workspace/qtquick/pointcloud/Qt3DPointcloudRenderer/bld/example/Qt3DPointcloudRenderer_example+0x4ed818)
    #1 0x7f75d32b86b8  (/opt/Qt5.7.0/5.7/gcc_64/lib/libQt53DRender.so.5+0x1536b8)

Thread T3 (Qt3DCore::QAspe) created by T0 here:
    #0 0x456b8d in pthread_create (/home/matt/workspace/qtquick/pointcloud/Qt3DPointcloudRenderer/bld/example/Qt3DPointcloudRenderer_example+0x456b8d)
    #1 0x7f75c81b3ba8 in QThread::start(QThread::Priority) (/opt/Qt5.7.0/5.7/gcc_64/lib/libQt5Core.so.5+0xaaba8)

SUMMARY: AddressSanitizer: alloc-dealloc-mismatch (/home/matt/workspace/qtquick/pointcloud/Qt3DPointcloudRenderer/bld/example/Qt3DPointcloudRenderer_example+0x51be90) in operator delete(void*)
==23217==HINT: if you don't care about these errors you may set ASAN_OPTIONS=alloc_dealloc_mismatch=0
==23217==ABORTING
```

**Comment by dabulla** [comment]
Thanks for the info. I noticed the segfault on exit too. This came with the update from Qt 5.6 to Qt 5.7 and it seems to happen only when the renderstate "PointSize" is used. Let's hope it's fixed in Qt 5.8.

**Comment by dabulla** [comment]
Issue is fixed in Qt 5.9.
